proto/api/v1/central_events_service.proto

```diff
@@ -0,0 +1,143 @@
+    // such as a particular image. In that case, the resource type will be filled here.
+    string resource_type                       = 7;
+
+    // Resource ID associated with the event. If a event refers to an underlying resource,
+    // the resource ID identifies the underlying resource.
+    string resource_id                         = 8;
+
+    // Occurrences associated with the event. When events may occur multiple times, the
+    // occurrences track the amount.
+    int64 num_occurrences                      = 9;
+
+    // Specifies the time when the event has last occurred.
+    google.protobuf.Timestamp last_occurred_at = 10;
+
+    // Specifies the time when the event has been created.
+    google.protobuf.Timestamp created_at       = 11;
+}
+
+// CentralEventType exposes the different types of events.
+enum CentralEventType {
+    CENTRAL_EVENT_TYPE_UNKNOWN     = 0;
+    CENTRAL_EVENT_TYPE_GENERIC     = 1;
+    CENTRAL_EVENT_TYPE_LOG_MESSAGE = 2;
+}
+
+// CentralEventLevel exposes the different levels of events.
+enum CentralEventLevel {
+    CENTRAL_EVENT_LEVEL_UNKNOWN = 0;
+    CENTRAL_EVENT_LEVEL_INFO    = 1;
+    CENTRAL_EVENT_LEVEL_SUCCESS = 2;
+    CENTRAL_EVENT_LEVEL_WARN    = 3;
+    CENTRAL_EVENT_LEVEL_DANGER  = 4;
+}
+
+message CentralEventsFilter {
+    // For filtering events starting from a specific timestamp, i.e. the lower boundary.
+    google.protobuf.Timestamp from  = 1;
+
+    // For filtering events up until a specific timestamp, i.e. the upper boundary.
+    google.protobuf.Timestamp until = 2;
+
+    // For filtering events from a specific domain.
+    string domain                   = 3;
+
+    // For filtering events associated with a specific resource type.
+    string resource_type            = 4;
+
+    // For filtering events based on their type.
+    CentralEventType type        = 5;
+
+    // For filtering events based on their level.
+    CentralEventLevel level      = 6;
+}
+
+message CountCentralEventsRequest {
+    // For filtering the events based on the requested fields.
+    CentralEventsFilter filter = 1;
+}
+
+message CountCentralEventsResponse {
+    // The total number of events after filtering and deduplication.
+    int64 count = 1;
+}
+
+message GetCentralEventResponse {
+    CentralEvent event = 1;
+}
+
+message ListCentralEventsRequest {
+    // For dividing the events response into chunks.
+    Pagination pagination         = 1;
+
+    // For filtering the events based on the requested fields.
+    CentralEventsFilter filter = 2;
+}
+
+message ListCentralEventsResponse {
+    repeated CentralEvent events = 1;
+}
+
+service CentralEventService {
+    // CountCentralEvents returns the number of events after filtered by requested fields.
+    rpc CountCentralEvents(CountCentralEventsRequest) returns (CountCentralEventsResponse) {
+        option (google.api.http) = {
+            get: "/v1/centraleventscount"
```

Comment:
(applies to all APIs):
My two cents on the naming of the API: I personally feel long words without separators quite hard to read within the API. There's precedent in other APIs where we do have quite long names, but I'd prefer `central-events` rather.

On the overall naming: while `Central` may be clear in the naming, if I was to call this API, I would be quite confused with the `central` being there. The API I'm calling _is_ already defined to be Central by the hostname I'm using for the API call, so duplicating this info feels _meh_ to me.

Briefly discussed this with @stehessel and there might be room to scope this API via e.g. `v1/platform/events`. Alternatively, there is also `/v1/product/usage`, but `product` feels too vague IMO since _technically everything_ is the product, so something like `platform` or `system` would be more in-line.

Ultimately, we should come to a common understanding rather sooner than later, I'd not be willing to ponder too long on the naming. We can use the sync on Friday to discuss this and come to a conclusion to avoid too many refactorings now. If everyone votes for `centralevents`, I'm willing to be convinced for that one.

cc: @rukletsov @pedrottimark